app/lib/katello/util/path_with_substitutions.rb

```diff
@@ -12,12 +12,35 @@ def initialize(path, substitutions)
         @substitutions = substitutions
         @path = path
         @resolved = []
+        create_rhel_eight_substitutions
+      end
+
+      def split_path
+        @split ||= @path.split('/').reject(&:blank?)
       end
 
       def substitutions_needed
         # e.g. if content_url = "/content/dist/rhel/server/7/$releasever/$basearch/kickstart"
         #      return ['releasever', 'basearch']
-        path.split('/').map { |word| word.start_with?('$') ? word[1..-1] : nil }.compact
+        split_path.map { |word| word.start_with?('$') ? word[1..-1] : nil }.compact
+      end
+
+      def create_rhel_eight_substitutions
+        @substitutions["basearch"] = rhel_eight_arch if rhel_eight?
+      end
+
+      def rhel_eight?
+        # url for RHEL8 repos can be either format:
+        # /content/dist/rhel8/8.0/x86_64/baseos/os/ OR
+        # /content/dist/layered/rhel8/x86_64/product
+        rhel8 = "rhel8"
+        split_path[2] == rhel8 || split_path[3] == rhel8
+      end
+
+      def rhel_eight_arch
+        # arch for RHEL8 repo paths is always the in the 5th position of the url i.e.
+        # /content/dist/rhel8/8.0/x86_64/baseos/os/ OR
+        split_path[4]
```

Comment:
This seems like it would be extremely fragile and could easily break.  Would it make more sense to have a list of arches and look for any of those, i feel like new arches will be introduced few times than the cdn will introduce some new path structure.